[Asset Inventory] Split URL State Management for Page Filters and Grouping fixes

opauloh opened the pull request:
## Summary

It closes #219580 and #219654

This PR adds separate URL state management to filters in the asset inventory page, so it doesn't conflict with the filter used in the Searchbar.

It also fixes an issue with the null grouping not filtering for the missing group field.

### Screenshot

https://github.com/user-attachments/assets/0c022d41-f68f-46df-bbe9-2c5918e0af44

**Null group fix**

<img width="1481" alt="image" src="https://github.com/user-attachments/assets/60010d40-f314-4823-bddc-17f828f175c0" />

**Asset Criticality Grouping**

<img width="1493" alt="image" src="https://github.com/user-attachments/assets/7a47742f-0d14-4643-91ac-ce3dd847b576" />

opauloh committed:
Adds page-level filters to asset inventory

Introduces page-level filters to the asset inventory.

This allows users to apply filters that persist across
different views and components within the asset inventory,
enhancing the filtering capabilities.

elasticmachine commented:
Pinging @elastic/kibana-cloud-security-posture (Team:Cloud Security)

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/299592)
* Commit: 07e8a6169828b162d5b942a771bcc04245a30d1d

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/299592#0196ad77-1d58-482b-991c-878d1f5619da)

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/299592","number":299592,"commit":"07e8a6169828b162d5b942a771bcc04245a30d1d"}],"number":299592}
buildkite-pr-comment-->

opauloh committed:
Initializes pageFilters as an empty array

Ensures `pageFilters` is initialized as an empty array
in `getDefaultQuery` to prevent potential issues
with undefined or missing filter values.

albertoblaz reviewed (approved):
👏

opauloh committed:
Adds assertion to filter application

Ensures that page filters are not applied as app filters,
preventing unintended filtering behavior.

opauloh commented on x-pack/solutions/security/plugins/security_solution/public/asset_inventory/hooks/use_asset_inventory_url_state/use_base_es_query.test.ts:
```diff
@@ -0,0 +1,129 @@
+
+    expect(filterManager.setAppFilters).toHaveBeenCalledWith(filters);
+    expect(queryString.setQuery).toHaveBeenCalledWith(query);
+    expect(notifications.toasts.addError).not.toHaveBeenCalled();
```
We actually want it not to be added to the filter manager, but good call, I added the assertion!

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/299766)
* Commit: 54877d3384c6a3992a2b385a47b72c29e72bd9e5

### Failed CI Steps
* [FTR Configs #93](https://buildkite.com/elastic/kibana-pull-request/builds/299766#0196b064-9be1-44ec-a99a-f8378a9f40e9)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/299766#0196b064-9be1-44ec-a99a-f8378a9f40e9) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/299766/jobs/0196b064-9be1-44ec-a99a-f8378a9f40e9/artifacts/0196b090-9b82-4677-a358-eab68231fdd6) FTR Configs #<span></span>93 / Dataset Quality Dataset quality handles user privileges User can read logs-* User can monitor some data streams shows underprivileged warning when size cannot be accessed for some data streams

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/10d46e5f5e8d331fb61e83eb52494e47d28ca5b6) | [after](https://github.com/elastic/kibana/commit/54877d3384c6a3992a2b385a47b72c29e72bd9e5) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 9.1MB | 9.1MB | +211.0B |



### History
* :broken_heart: [Build #299592](https://buildkite.com/elastic/kibana-pull-request/builds/299592) failed 07e8a6169828b162d5b942a771bcc04245a30d1d

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/299766","number":299766,"commit":"54877d3384c6a3992a2b385a47b72c29e72bd9e5"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/299592","number":299592,"commit":"07e8a6169828b162d5b942a771bcc04245a30d1d"}],"number":299766}
buildkite-pr-comment-->

opauloh committed:
Removing pageFilters from filter manager

This simplifies the logic and prevents issues arising from redundant filter management.

opauloh committed:
Merge remote-tracking branch 'elastic/main' into asset-inventory/page-filters

opauloh committed:
handling undefined pageFilter

opauloh committed:
Extends query filter to support bool queries

Allows the asset inventory table to handle boolean `should` queries
in addition to `match_phrase` queries when filtering data.

opauloh committed:
Displays criticality badge in asset inventory

Adds an asset criticality badge to the asset inventory group
panel to improve the display of asset criticality information.

This enhances the user experience by providing a visual
representation of asset criticality within the inventory.

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/299948)
* Commit: 9704de00b4f7cccc777416083f1d23d9a63e9bcc

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/96d1692ae20172ea3d8a5f7062c55c10d38ed9a2) | [after](https://github.com/elastic/kibana/commit/9704de00b4f7cccc777416083f1d23d9a63e9bcc) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 9.1MB | 9.1MB | +252.0B |



### History
* :yellow_heart: [Build #299766](https://buildkite.com/elastic/kibana-pull-request/builds/299766) was flaky 54877d3384c6a3992a2b385a47b72c29e72bd9e5
* :broken_heart: [Build #299592](https://buildkite.com/elastic/kibana-pull-request/builds/299592) failed 07e8a6169828b162d5b942a771bcc04245a30d1d

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/299948","number":299948,"commit":"9704de00b4f7cccc777416083f1d23d9a63e9bcc"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/299766","number":299766,"commit":"54877d3384c6a3992a2b385a47b72c29e72bd9e5"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/299592","number":299592,"commit":"07e8a6169828b162d5b942a771bcc04245a30d1d"}],"number":299948}
buildkite-pr-comment-->
